programs/bpf_loader/src/lib.rs

```diff
@@ -248,11 +254,66 @@ pub fn create_executor_from_account(
         false, /* reject_deployment_of_broken_elfs */
     )?;
     if let Some(mut tx_executor_cache) = tx_executor_cache {
-        tx_executor_cache.set(*program.get_key(), executor.clone(), false);
+        tx_executor_cache.set(*program.get_key(), executor.clone(), false, false);
     }
     Ok((executor, Some(create_executor_metrics)))
 }
 
+macro_rules! deploy_program {
+    ($invoke_context:expr, $use_jit:expr, $program_id:expr,
+     $program_account:expr, $programdata_account:expr, $new_programdata:expr,
+     $first_drop:expr, $second_drop:expr $(,)?) => {{
+        let delay_visibility_of_program_deployment = $invoke_context
+            .feature_set
+            .is_active(&delay_visibility_of_program_deployment::id());
+
+        if delay_visibility_of_program_deployment {
```

Comment:
It's confusing to me that we call `create_executor_from_account` here is `delay_visibility_of_program_deployment` is true, but then `create_executor_from_account` passes false for `delay_visibility_of_program_deployment` to `set()`